lib/Kernel.php

```diff
@@ -236,10 +237,14 @@ protected function initializeContainer()
             // check if container still exists at this point as it could already
             // be cleared (e.g. when running tests which boot multiple containers)
             try {
-                $this->getContainer()->get('event_dispatcher')->dispatch(new GenericEvent(), SystemEvents::SHUTDOWN);
-            } finally {
-                \Pimcore::shutdown();
+                $container = $this->getContainer();
+            } catch (\LogicException) {
+                // Container is cleared. Allow tests to finish.
             }
+            if (isset($container) && $container instanceof ContainerInterface) {
+                $container->get('event_dispatcher')->dispatch(new GenericEvent(), SystemEvents::SHUTDOWN);
```

Comment:
Shouldn't we move that inside the `try` block?

Reply:
I wouldn't do that because you'll lose exceptions from container and dispatcher. We shouldn't catch them.